pyiron_atomistics/atomistics/structure/factory.py

```diff
@@ -407,3 +416,106 @@ def from_pymatgen(pymatgen_obj):
+            length_terrace (int): The length of the terrace along the kink direction in atoms eg., 3
+            length_step (int): The length of the step along the step direction in atoms eg., 3
+            length_kink (int): The length of the kink along the kink direction in atoms eg., 1
+
+
+        Returns:
+            high_index_surface: The high miller index surface which can be used to create slabs
+            fin_kink_orientation: The kink orientation lying in the terrace
+            fin_step_orientation: The step orientation lying in the terrace
+        """
+        terrace_orientation = terrace_orientation if terrace_orientation is not None else [1, 1, 1]
+        step_orientation = step_orientation if step_orientation is not None else [1, 1, 0]
+        kink_orientation = kink_orientation if kink_orientation is not None else [1, 1, 1]
+        step_down_vector = step_down_vector if step_down_vector is not None else [1, 1, 0]
+
+        basis = self.crystal(element=element, bravais_basis=crystal_structure, lattice_constant=lattice_constant)
+        sym = basis.get_symmetry()
+        eqvdirs = np.unique(np.matmul(sym.rotations[:], (np.array(step_orientation))), axis=0)
+        eqvdirk = np.unique(np.matmul(sym.rotations[:], (np.array(kink_orientation))), axis=0)
+        eqvdirs_ind = np.where(np.dot(np.squeeze(eqvdirs), terrace_orientation) == 0)[0]
+        eqvdirk_ind = np.where(np.dot(np.squeeze(eqvdirk), terrace_orientation) == 0)[0]
+        if len(eqvdirs_ind) == 0:
+            raise ValueError('Step orientation vector should lie in terrace.\
+            For the given choice I could not find any symmetrically equivalent vector that lies in the terrace.\
+            please change the stepOrientation and try again')
+        if len(eqvdirk_ind) == 0:
+            raise ValueError('Kink orientation vector should lie in terrace.\
+            For the given choice I could not find any symmetrically equivalent vector that lies in the terrace.\
+            please change the kinkOrientation and try again')
+        temp = (np.cross(np.squeeze(eqvdirk[eqvdirk_ind[0]]), np.squeeze(eqvdirs))).tolist().index(terrace_orientation)
+        fin_kink_orientation = eqvdirk[eqvdirk_ind[0]]
+        fin_step_orientation = eqvdirs[temp]
+        vec1 = (np.asanyarray(fin_step_orientation).dot(length_step)) + \
+               (np.asanyarray(fin_kink_orientation).dot(length_kink))
+        vec2 = (np.asanyarray(fin_kink_orientation).dot(length_terrace)) + step_down_vector
+        high_index_surface = np.cross(np.asanyarray(vec1), np.asanyarray(vec2))
+        high_index_surface = np.array(high_index_surface / np.gcd.reduce(high_index_surface), dtype=int)
+
+        return high_index_surface, fin_kink_orientation, fin_step_orientation
+
+    def high_index_surface(self, element='Ni', crystal_structure='fcc', lattice_constant=3.526,
```

Comment:
To be consistent with other methods in this class, I think there shouldn't be default values for `element`, `crystal_structure`, and `lattice_constant` but rather require that the user specify these values.